chaco/axis.py

```diff
@@ -244,23 +244,38 @@ def overlay(self, component, gc, view_bounds=None, mode="normal"):
         """
         if not self.visible:
             return
-        self._draw_component(gc, view_bounds, mode, component)
+
+        if not self._cache_valid:
+            if component is not None:
+                self._calculate_geometry_overlay(component)
+            else:
+                self._calculate_geometry()
+            self._compute_tick_positions(gc, component)
+            self._compute_labels(gc)
+
+        with gc:
+            # slight optimization: if we set the font correctly on the
+            # base gc before handing it in to our title and tick labels,
+            # their set_font() won't have to do any work.
+            gc.set_font(self.tick_label_font)
+
+            if self.axis_line_visible:
+                self._draw_axis_line(
+                    gc, self._origin_point, self._end_axis_point
+                )
+            if self.title:
+                self._draw_title(gc)
+
+            self._draw_ticks(gc)
+            self._draw_labels(gc)
+
+        self._cache_valid = True
 
     def _draw_overlay(self, gc, view_bounds=None, mode="normal"):
         """Draws the overlay layer of a component.
 
         Overrides PlotComponent.
         """
-        self._draw_component(gc, view_bounds, mode)
-
-    def _draw_component(
```

Comment:
I would prefer leaving the `_draw_component` method as it is instead of duplicating it in `overlay` and `_draw_overlay`.

we could simply update the outdated the docstring in this case.

Reply:
never mind. let's make `overlay` use `_draw_overlay` like you mentioned in the earlier comment